lib/routes/routeBackbeat.js

```diff
@@ -482,6 +482,7 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
             // Retrieve the null version id from the object metadata.
             versionId = objMd && objMd.versionId;
             if (!versionId) {
+                omVal.isNull = true;
                 if (versioning) {
                     // If the null version does not have a version id, it is a current null version.
```

Comment:
In all the cases I could think of the non-versioned was converted to a version + master the next time an object was PUT.

Reply:
OK, but it's nevertheless incorrect to update the master if the version is not the current version, we need to guard against this.

Reply:
I've poked around a bit more and it doesn't seem to be able to have a non-versioned (internal version) object that is also not the master in a versioned bucket. In every case the object would have a versionId so the check on line 484 ensures this is the case.

Reply:
Yes that's right, the nonversioned object is appended the "infinite" version ID when new versions are created on top.

Reply:
> Out of context, this comment line looks wrong since a nonversioned object is not necessarily the current version (doesn't mean the code is wrong).

So regarding my statement above, indeed it's the current version so the code comment is right, but we could maybe go further and say that it can only be a *unique* version because as soon as new versions are created, Cloudserver converts the nonversioned object to a versioned one with an infinite version ID and the `isNull` flag.